packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1426,6 +1474,85 @@ export class TransactionController extends BaseController<
 
     return dappSuggestedGasFees;
   }
+
+  /**
+   * Validates and adds external provided transaction to state.
+   *
+   * @param transactionMeta - Nominated external transaction to be added to state.
+   */
+  private async addExternalTransaction(transactionMeta: TransactionMeta) {
+    const { networkId, chainId } = this.getChainAndNetworkId();
+    const fromAddress = transactionMeta?.transaction?.from;
+    const confirmedTxs = this.state.transactions.filter(
```

Comment:
As we use `this.state.transactions` in a few places, is it worth giving it a local variable via a dereference?

Reply:
Fixed in https://github.com/MetaMask/core/pull/1625/commits/3e69824b2e6fac00c62ef54fffc485137248f95f